src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -0,0 +1,586 @@
+        {
+            _originalFontName = familyName;
+
+            SetFontFamily(new FontFamily(StripVerticalName(familyName), createDefaultOnFail: true));
+            Initialize(_fontFamily, emSize, style, unit, gdiCharSet, gdiVerticalFont);
+        }
+
+        /// <summary>
+        /// Initializes this object's fields.
+        /// </summary>
+        private void Initialize(FontFamily family, float emSize, FontStyle style, GraphicsUnit unit, byte gdiCharSet, bool gdiVerticalFont)
+        {
+            if (family == null)
+            {
+                throw new ArgumentNullException(nameof(family));
+            }
+
+            if (float.IsNaN(emSize) || float.IsInfinity(emSize) || emSize <= 0)
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidBoundArgument, nameof(emSize), emSize, 0, "System.Single.MaxValue"), nameof(emSize));
+            }
+            
+            // Libgdiplus doesn't perform argument validation for the unit.
+            if (unit < GraphicsUnit.World || unit > GraphicsUnit.Millimeter || unit == GraphicsUnit.Display)
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidEnumArgument, nameof(unit), unit, nameof(GraphicsUnit)), nameof(unit));
+            }
+
+            int status;
+
+            _fontSize = emSize;
+            _fontStyle = style;
+            _fontUnit = unit;
+            _gdiCharSet = gdiCharSet;
+            _gdiVerticalFont = gdiVerticalFont;
+
+            if (_fontFamily == null)
+            {
+                // GDI+ FontFamily is a singleton object.
+                SetFontFamily(new FontFamily(family.NativeFamily));
+            }
+
+            if (_nativeFont == IntPtr.Zero)
+            {
+                CreateNativeFont();
+            }
+
+            // Get actual size.
+            status = SafeNativeMethods.Gdip.GdipGetFontSize(new HandleRef(this, _nativeFont), out _fontSize);
+            SafeNativeMethods.Gdip.CheckStatus(status);
+        }
+
+        private static bool IsVerticalName(string familyName) => familyName?.Length > 0 && familyName[0] == '@';
+
+        private static string StripVerticalName(string familyName)
+        {
+            if (familyName?.Length > 1 && familyName[0] == '@')
+            {
+                return familyName.Substring(1);
+            }
+
+            return familyName;
+        }
+
+        private void SetFontFamily(FontFamily family)
+        {
+            _fontFamily = family;
+
+            // GDI+ creates ref-counted singleton FontFamily objects based on the family name so all managed 
+            // objects with same family name share the underlying GDI+ native pointer. The unmanged object is
+            // destroyed when its ref-count gets to zero.
+            // Make sure this.fontFamily is not finalized so the underlying singleton object is kept alive.
```

Comment:
Is all of this true with libgdiplus as well?  Otherwise this is creating a leak.

Reply:
Actually not, only for Generic font families - fixed